presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -167,7 +168,7 @@ public ConnectorPageSource createPageSource(
                 split.getLength(),
                 split.getFileFormat(),
                 regularColumns,
-                table.getPredicate());
+                table.getPredicate().intersect(dynamicFilter.getCurrentPredicate().transform(IcebergColumnHandle.class::cast).simplify(100)));
```

Comment:
if you passed `dynamicFilter` to `IcebergPageSource` you could easily block on DF in `io.prestosql.spi.connector.ConnectorPageSource#isBlocked`. However, this should be behind feature toggle

Reply:
It seems from the code that stripe pruning happens before `IcebergPageSource` is created in `createDataPageSource -> createOrcPageSource -> reader.createRecordReader -> OrcRecordReader`. So we would miss that even if we block in IcebergPageSource.
I think row group pruning could still be accomplished by blocking on DF in IcebergPageSource if `dynamicFilter` is pushed into `StripeReader` as well. Not sure if the changes are worth doing though.
Please correct if I'm missing something.

Reply:
We could probably create `createDataPageSource` in `IcebergPageSource` in a lazy way (when DF is ready). This way we don't allocate resources until DF is ready (I'm not sure it's big of an issue though)